src/components/AddressPicker/AddressPicker.js

```diff
@@ -1,100 +1,117 @@
 
+import checkAddressValid from 'utils/wallets/checkAddressValid'
 import JPicker, { JPickerFullItem } from 'components/base/JPicker'
 
-import Current from './Current/AddressPickerCurrent'
+import AddressPickerCurrent from './Current'
 
 type Props = {|
-  addresses: Array<AddressPickerAddress>,
-  selectedAddress: Address,
-  onSelect: (address: Address) => void,
-  isDisabled: boolean,
-  errorMessage: string,
+  +onSelect: (address: Address) => void,
+  +addressNames: AddressNames,
+  +errorMessage: string,
+  +selectedAddress: Address,
+  +isDisabled: boolean,
 |}
 
 type ComponentState = {|
-  filter: string,
+  +searchQuery: string,
 |}
 
+function filterAddressNames(addressNames: AddressNames, searchQuery: string): AddressNames {
+  return !searchQuery ? addressNames : Object.keys(addressNames).reduce((
+    result: AddressNames,
+    address: Address,
+  ): AddressNames => {
+    const name: ?string = addressNames[address]
+
+    if (!name) {
+      return result
+    }
+
+    const isAddressFound: boolean = (address.toLowerCase() === searchQuery)
+    const isNameFound: boolean = (name.toLowerCase().indexOf(searchQuery) !== -1)
+    const isFound: boolean = (isAddressFound || isNameFound)
+
+    return !isFound ? result : {
+      ...result,
+      [address]: name,
+    }
+  }, {})
+}
+
 class AddressPicker extends Component<Props, ComponentState> {
   static defaultProps = {
-    addresses: [],
-    isDisabled: false,
     errorMessage: '',
+    isDisabled: false,
   }
 
   constructor(props: Props) {
     super(props)
 
     this.state = {
-      filter: '',
+      searchQuery: '',
     }
   }
 
-  onFilterChange = (filter: string) => {
-    this.setState({ filter }, () => {
-      if (checkAddressValid(filter)) {
-        this.props.onSelect(filter)
+  changeSearchQuery = (searchQuery: string) => {
+    this.setState({ searchQuery }, () => {
+      if (checkAddressValid(searchQuery)) {
+        this.props.onSelect(searchQuery)
       }
     })
   }
 
-  onOpen = () => {
-    // reset filter
-    this.setState({ filter: '' })
-  }
+  cleanSearchQuery = () => this.setState({ searchQuery: '' })
 
   render() {
     const {
-      addresses,
-      selectedAddress,
       onSelect,
-      isDisabled,
+      addressNames,
+      selectedAddress,
       errorMessage,
-    } = this.props
-
-    const {
-      filter,
-    } = this.state
-
-    const activeAddress = addresses.find(({ address }) => address === selectedAddress)
+      isDisabled,
+    }: Props = this.props
 
-    const filterStr = filter.trim().toLowerCase()
-    const filtered = filterStr
-      ? addresses.filter(({ address, title }) =>
-        address.toLowerCase() === filterStr ||
-          title.toLowerCase().indexOf(filterStr) !== -1
-      )
-      : addresses
+    const { searchQuery }: ComponentState = this.state
+    const query: string = searchQuery.trim().toLowerCase()
+    const filteredAddressNames: AddressNames = filterAddressNames(addressNames, query)
```

Comment:
Maybe you can use regex for filtering? It can be more laconic and flexible approach.